src/adaptors/pancakeswap-v3/cakeReward.js

```diff
@@ -0,0 +1,125 @@
+const abiMcV3 = require('./masterchefv3.json');
+const utils = require('../utils');
+const sdk = require('@defillama/sdk');
+const bn = require('bignumber.js');
+
+const MASTERCHEF_ADDRESS = '0x556B9306565093C855AEA9AE92A594704c2Cd59e';
+
+const baseUrl = 'https://api.thegraph.com/subgraphs/name';
+const chains = {
+  ethereum: `${baseUrl}/pancakeswap/exchange-v3-eth`,
+  bsc: `${baseUrl}/pancakeswap/exchange-v3-bsc`,
+};
+
+const getCakeAprs = async (chain) => {
+  const poolLength = await sdk.api.abi
+    .call({
+      abi: abiMcV3.find((m) => m.name === 'poolLength'),
+      target: MASTERCHEF_ADDRESS,
+      params: chain,
+    })
+    .then((o) => o.output);
+  const totalAllocPoint = await sdk.api.abi
+    .call({
+      abi: abiMcV3.find((m) => m.name === 'totalAllocPoint'),
+      target: MASTERCHEF_ADDRESS,
+      params: chain,
+    })
+    .then((o) => o.output);
+  const latestPeriodCakePerSecond = await sdk.api.abi
+    .call({
+      abi: abiMcV3.find((m) => m.name === 'latestPeriodCakePerSecond'),
+      target: MASTERCHEF_ADDRESS,
+      params: chain,
+    })
+    .then((o) => o.output);
+
+  const cakePerSecond = new bn(latestPeriodCakePerSecond.toString())
+    .div(1e18)
+    .div(1e12)
+    .toString();
+
+  const poolInfoCalls = Array.from({ length: poolLength + 1 })
+    .map((_, i) => i)
+    .filter((i) => i !== 0)
+    .map((i) => {
+      return {
+        target: MASTERCHEF_ADDRESS,
+        params: i,
+      };
+    });
+
+  const poolInfos = await sdk.api.abi
+    .multiCall({
+      abi: abiMcV3.find((m) => m.name === 'poolInfo'),
+      calls: poolInfoCalls,
+      chain,
+    })
+    .then((o) => o.output);
+
+  const { cakePrice } = await getBaseTokensPrice();
+
+  // TODO: Here a little too complex to get the total staked liquidity in MasterChef `in-range`, can use api to get it instead?
+  // https://github.com/pancakeswap/pancake-frontend/blob/develop/apps/web/src/pages/api/v3/%5BchainId%5D/farms/liquidity/%5Baddress%5D.ts
+  // https://pancakeswap.finance/api/v3/1/farms/liquidity/0x1ac1A8FEaAEa1900C4166dEeed0C11cC10669D36
+  // by lp address
+  const tvls = {};
+
+  // TODO: convert it into usd
+  // by lp address
+  const tvlsUSD = {};
```

Comment:
need help on getting staked tvl on MasterChef v3. 
detail can check the links

Reply:
henlo, if too complex then api is fine. for usd stuff u can use our coins api